tests/types/test_schema.py

```diff
@@ -247,33 +250,113 @@ def test_schema_inference_on_dictionary(dict_of_ndarrays):
         _infer_schema({"x": [1]})
 
 
-def test_schema_inference_on_numpy_array(pandas_df_with_all_types):
+def test_schema_inference_on_basic_numpy(pandas_df_with_all_types):
     for col in pandas_df_with_all_types:
         data = pandas_df_with_all_types[col].to_numpy()
         schema = _infer_schema(data)
         assert schema == Schema([TensorSpec(type=data.dtype, shape=(-1,))])
 
-    # test boolean
-    schema = _infer_schema(np.array([True, False, True], dtype=np.bool_))
-    assert schema == Schema([TensorSpec(np.dtype(np.bool_), (-1,))])
-
-    # test bytes
-    schema = _infer_schema(np.array([bytes([1])], dtype=np.bytes_))
-    assert schema == Schema([TensorSpec(np.dtype("S1"), (-1,))])
 
-    # test (u)ints
-    for t in [np.uint8, np.int8, np.uint16, np.int16, np.uint32, np.int32, np.uint64, np.int64]:
-        schema = _infer_schema(np.array([1, 2, 3], dtype=t))
-        assert schema == Schema([TensorSpec(np.dtype(t), (-1,))])
+def test_all_numpy_dtypes():
+    def test_dtype(nparray, dtype):
+        schema = _infer_schema(nparray)
+        assert schema == Schema([TensorSpec(np.dtype(dtype), (-1,))])
```

Comment:
This comparison should fail for strings and bytes right? Did you forget to call clean_tensor_dtype?

Reply:
The `dtype` used to create the TensorSpec does not contain length information. It comes from the strings stored in the datatype arrays:
`bytes_ = ["bytes_", "string_"]
    str_ = ["str_", "unicode_"]`

Reply:
oh, right, I missed that :)